File: SjoslagetBackend/Models/BookingOverviewItem.cs
using System;

namespace Accidis.Sjoslaget.WebService.Models
{
	public sealed class BookingOverviewItem
	{
		public Guid Id { get; set; }
		public string Reference { get; set; }
		public string SubCruise { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string Lunch { get; set; }
		public decimal TotalPrice { get; set; }
		public decimal AmountPaid { get; set; }
		public int NumberOfCabins { get; set; }
		public bool IsLocked { get; set; }
		public DateTime Updated { get; set; }
	}
}

File: SjoslagetFrontend/lib/model/booking_overview_item.dart
import 'package:decimal/decimal.dart';
import 'package:frontend_shared/model/booking_payment_model.dart';
import 'package:quiver/strings.dart' as str show isEmpty;

import 'json_field.dart';

class BookingOverviewItem extends BookingPaymentModel {
  final String reference;
  final String subCruise;
  final String firstName;
  final String lastName;
  final String lunch;
  final int numberOfCabins;
  final bool isLocked;
  final DateTime updated;

  BookingOverviewItem(this.reference, this.subCruise, this.firstName, this.lastName, this.lunch, Decimal totalPrice, Decimal amountPaid,
      this.numberOfCabins, this.isLocked, this.updated)
      : super(totalPrice, amountPaid);

  factory BookingOverviewItem.fromMap(Map<String, dynamic> json) => BookingOverviewItem(
      json[REFERENCE],
      json[SUBCRUISE],
      json[FIRSTNAME],
      json[LASTNAME],
      json[LUNCH],
      Decimal.parse(json[TOTAL_PRICE].toString()),
      Decimal.parse(json[AMOUNT_PAID].toString()),
      json[NUMBER_OF_CABINS],
      json[IS_LOCKED],
      DateTime.parse(json[UPDATED]));

  String get lunchFormatted => str.isEmpty(lunch) ? '' : '$lunch:00';
}
